src/plugins/discord/fetcher.test.js

```diff
@@ -0,0 +1,268 @@
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      await fetcher.channels("1");
+      expect(fetch.mock.calls[0]).toEqual(["/guilds/1/channels"]);
+    });
+
+    it("handles response", async () => {
+      const testChannel = (id: string) => ({
+        id: id,
+        name: "testChannelName",
+        type: 0,
+      });
+
+      const testChannelObj = (id: Snowflake): Channel => ({
+        id: id,
+        name: "testChannelName",
+        type: "GUILD_TEXT",
+      });
+
+      const response = [testChannel("1")];
+      const expected = [testChannelObj("1")];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      const data = await fetcher.channels("0");
+      expect(data).toEqual(expected);
+    });
+  });
+
+  describe("fetch members", () => {
+    const testMember = (userId: number) => ({
+      user: {
+        id: userId,
+        username: "username",
+        discriminator: "disc",
+        bot: true,
+      },
+      nick: "nickname",
+      roles: ["test role"],
+    });
+
+    const options = () => ({
+      membersLimit: 2,
+      messagesLimit: 100,
+      reactionsLimit: 100,
+    });
+
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, options());
+      await fetcher.members("1", "0");
+      expect(fetch.mock.calls[0]).toEqual([
+        "/guilds/1/members?after=0&limit=2",
+      ]);
+    });
+
+    it("handles response", async () => {
+      const response = [testMember(1)];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {results} = await fetcher.members("1", "0");
+      expect(results).toEqual(response);
+    });
+
+    it("returns correct endCursor", async () => {
+      const response = [testMember(1), testMember(2)];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {endCursor} = (await fetcher.members("1", "0")).pageInfo;
+      expect(endCursor).toEqual(2);
+    });
+
+    describe("next page", () => {
+      const testHasNextPage = async (nextPage) => {
+        const response = [testMember(1)];
+        const fetch = jest.fn(() => Promise.resolve(response));
+        const fetcher = new Fetcher(fetch, {
+          membersLimit: nextPage ? 1 : 2,
+          messagesLimit: 100,
+          reactionsLimit: 100,
+        });
+        const {hasNextPage} = (await fetcher.members("1", "0")).pageInfo;
+        return hasNextPage;
+      };
+
+      it("next page = true", async () => {
+        expect(await testHasNextPage(true)).toBe(true);
+      });
+
+      it("next page = false", async () => {
+        expect(await testHasNextPage(false)).toBe(false);
+      });
```

Comment:
For readability, I believe it would be better to side with copy pasted code if it makes your test more straightforward. If you've seen examples of how I write tests with [Given, When, Then](https://en.wikipedia.org/wiki/Given-When-Then) it would work out something like this:

Given membersLimit=1 and 1 member
When fetching members
Then hasNextPage should be true

Given membersLimit=2 and 1 member
When fetching members
Then hasNextPage should be false